### PR TITLE
[4871] [studio] Add a publishing submission type (user, system, ...)

### PR DESCRIPTION
**Pull request by dejan-brkic**
Publishing submission type feature - design

### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/4871

### CODE REVIEW

File: src/main/api/2/studio-api-2.yaml
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: Remove `mandatory and instead calculate this on the backend and call it internally `mandatoryDependencies`.

File: src/main/api/2/studio-api-2.yaml
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: Rename `optional` to `optionalDependencies`

File: src/main/api/2/studio-api-2.yaml
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: Rename `environment` to `publishingTarget`

File: src/main/api/2/studio-api-2.yaml
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: Rename `sendEmail` to `sendEmailNotifications`

File: src/main/api/2/studio-api-2.yaml
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: Change to: `if true, send email notifications`

File: src/main/api/2/studio-api-2.yaml
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: Change `publishComment` to `reviewerComment`

File: src/main/api/2/studio-api-2.yaml
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: Reviewer's comment

File: src/main/api/2/studio-api-2.yaml
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: Rename to `deletionComment`

File: src/main/api/2/studio-api-2.yaml
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: deletion comment by the user performing the delete

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/api/2/studio-api-2.yaml
Comment (rart) [reply src/main/api/2/studio-api-2.yaml]: What's the purpose of splitting "optional" dependencies out? (i.e. as opposed to send a single list of paths)

File: src/main/api/2/studio-api-2.yaml
Comment (rart) [inline src/main/api/2/studio-api-2.yaml:6367]: Is `schedule` optional? Can it receive empty string or not be in the request at all when schedule is now or what should the value be when submitted for `now`?

File: src/main/api/2/studio-api-2.yaml
Comment (rart) [inline src/main/api/2/studio-api-2.yaml]: On the UI, we only have one field in the dialog to enter a comment **and** the backend doesn't give us the previous submission comment to resend it on approval — nor do I see much point to it. Perhaps I'm misinterpreting, don't see why two different comments.

It helps to keep the two APIs congruent, so keeping same name for the comment on both APIs good. Perhaps just `comment`?

File: src/main/api/2/studio-api-2.yaml
Comment (rart) [inline src/main/api/2/studio-api-2.yaml]: On the UI, we only have one field in the dialog to enter a comment **and** the backend doesn't give us the previous submission comment to resend it on approval — nor do I see much point to it. Perhaps I'm misinterpreting, don't see why two comments on the same endpoint.

It helps to keep the two APIs congruent, so keeping same name for the comment on both APIs would be good. Perhaps just `comment`?

File: src/main/api/2/studio-api-2.yaml
Comment (rart) [inline src/main/api/2/studio-api-2.yaml:6360]: What's the purpose of splitting "optional" dependencies out? (i.e. as opposed to send a single list of paths).
Both items & optional dependencies are human-choice (i.e. people must check them to get submitted).

File: src/main/api/2/studio-api-2.yaml
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: Rename environment

File: src/main/api/2/studio-api-2.yaml
Comment (sumerjabri) [reply src/main/api/2/studio-api-2.yaml:6367]: @dejan-brkic this can be omitted since it's optional and that means `now`, please confirm

File: src/main/api/2/studio-api-2.yaml
Comment (rart) [inline src/main/api/2/studio-api-2.yaml]:
```suggestion
                - publishingTarget
```

File: src/main/api/2/studio-api-2.yaml
Comment (dejan-brkic) [reply src/main/api/2/studio-api-2.yaml:6367]: I don't want to mix string values with dates. empty means now

File: src/main/api/2/studio-api-2.yaml
Comment (dejan-brkic) [reply src/main/api/2/studio-api-2.yaml]: Per API v1 docs, old API has 2 fields 
https://docs.craftercms.org/en/4.0/developers/projects/studio/api/workflow/go-live.html

File: src/main/api/2/studio-api-2.yaml
Comment (sumerjabri) [reply src/main/api/2/studio-api-2.yaml:6367]: Omitted = now

**Review by sumerjabri — CHANGES_REQUESTED** [review]
Please split `publish` into:
`publish` for direct publishing without an approval step/reviewer comment.
`approve` for publishing a set of items that have been submitted and are now being approved/published.

Please add `reject` to reject a submission.

File: src/main/api/2/studio-api-2.yaml
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml:6485]: @russdanner would the approver _add_ to the list of published items/package?

File: src/main/api/2/studio-api-2.yaml
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: Remove this.

File: src/main/api/2/studio-api-2.yaml
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml:6548]: What's the difference between reason and comment?

File: src/main/api/2/studio-api-2.yaml
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: Rename to `comment` so it's consistent with the rest.

**Review by sumerjabri — CHANGES_REQUESTED** [review]

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**
